src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3840,3 +4015,46 @@ class CallStatementSegment(BaseSegment):
             ),
         ),
     )
+
+
+@snowflake_dialect.segment(replace=True)
+class OrderByClauseSegment(BaseSegment):
+    """A `ORDER BY` clause like in `SELECT`."""
+
+    type = "orderby_clause"
+    match_grammar = StartsWith(
+        Sequence("ORDER", "BY"),
+        terminator=OneOf(
+            "LIMIT",
+            "HAVING",
+            "QUALIFY",
+            # For window functions
+            "WINDOW",
+            Ref("FrameClauseUnitGrammar"),
+            "SEPARATOR",
+            "MEASURES",
+        ),
+    )
+    parse_grammar = Sequence(
```

Comment:
Similar to https://github.com/sqlfluff/sqlfluff/pull/2784#discussion_r819567281 I think we should replace this with:

```py
parse_grammar = ansi_dialect.get_segment("OrderByClauseSegment").parse_grammar.copy()
```

Reply:
Good shout, will do!